packages/thirdweb/src/wallets/in-app/core/wallet/index.ts

```diff
@@ -23,13 +28,22 @@ export type InAppWalletAuth = "email" | "phone" | InAppWalletSocialAuth;
  * @internal
  */
 export async function connectInAppWallet(
-  options: InAppWalletConnectionOptions,
+  options: WalletConnectionOption<"inApp">,
+  createOptions: CreateWalletArgs<"inApp">[1],
 ): Promise<[Account, Chain]> {
   const { authenticate } = await import("../authentication/index.js");
 
   const authResult = await authenticate(options);
   const authAccount = await authResult.user.wallet.getAccount();
 
+  if (createOptions?.accountAbstraction) {
+    return convertToSmartAccount({
+      chain: options.chain || createOptions.accountAbstraction.chain,
```

Comment:
Any risk of this leading to unexpected behavior if the users specifies one chain on options and another in accountAbstraction (they shouldn't, but if they're confused)?

Reply:
yeah, been thinking of renaming the 'creation' chain to 'defaultChain' instead for clarity but didn't feel like it was worth the breaking change. lmk what you think

Reply:
yeah probably not worth it